docs/apis/subsystems/availability/index.md

```diff
@@ -0,0 +1,133 @@
+
+#### Course sections
+
+Some availability information can be accessed from an instance of the `section_info` class, specifically in the `uservisible` property.
+
+This property considers a range of factors to indicate whether the activity should be visible to that user, including:
+
+- whether the activity was hidden by a teacher
+- whether the activity is available based on the availability API
+
+The `section_info` object also includes an `availableinfo` property which provides HTML-formatted information to explain to the user why they cannot access the activity.
+
+:::warning Checking sections and activities
+
+The `uservisible` check for an _activity_ automatically includes all relevant checks for the section that the activity is placed in.
+
+You **do not** need to check visibility and availability for _both_ the section and the activity.
+
+:::
+
+#### Accessing information for a different user
+
+The availability information in both the `cm_info` and `section_info` classes is calculated  for the current user. You can also obtain them for a different user by passing a user ID to `get_fast_modinfo`, although be aware that doing this repeatedly for different users will be slow.
+
+### Display a list of users who may be able to access the current activity
+
+Sometimes you need to display a list of users who may be able to access the current activity.
+
+While you could use the above approach for each user, this would be slow and also is generally not what you require. For example, if you have an activity such as an assignment which is set to be available to students until a certain date, and if you want to display a list of potential users within that activity, you probably don't want to make the list blank immediately the date occurs.
+
+The system divides availability conditions into two types:
+
+- Applied to user lists, including:
+  - User group
+  - User grouping
+  - User profile conditions
+- Not applied to user lists, including:
+  - The current date
+  - completion
+  - grade
+
+In general, the conditions which we expect are likely to change over time (such as dates) or as a result of user actions (such as grades) are not applied to user lists.
+
+If you have a list of users (for example you could obtain this using one of the 'get enrolled users' functions), you can filter it to include only those users who are allowed to see an activity with this code:
+
+```php
+$info = new \core_availability\info_module($cm);
+$filtered = $info->filter_user_list($users);
+```
+
+:::note
+
+The above example does not include the `$cm->visible` setting, nor does it take into account the `viewhiddenactivities` setting.
+
+:::
+
+## Using availability conditions in other areas
+
+The availability API is provided for activities (course-modules) and sections. It is also possible to use it in other areas such as _within_ a module. See [Availability API for items within a module](https://docs.moodle.org/dev/Availability_API_for_items_within_a_module).
```

Comment:
Shouldn't this page (https://docs.moodle.org/dev/Availability_API_for_items_within_a_module) be migrated too here (or, at least, create a separate issue for it, pointing to the expected path to place it)?

Reply:
Possibly but I've pinged Sam Marshall to ask him about this becuase I can't understand what it is related to and how it compares to these docs.

Reply:
EIther way, let's handle that separately. Perfect is the enemy of good.

Reply:
I think they are different pages:
- you meant https://docs.moodle.org/dev/Conditional_activities_API (which was mentioned in your comment)
- and I was referring to https://docs.moodle.org/dev/Availability_API_for_items_within_a_module, which I think it's related to the Availability API